pkg/operator/ceph/cluster/osd/key_rotation.go

```diff
@@ -196,6 +196,7 @@ func (c *Cluster) getKeyRotationPodTemplateSpec(osdProps osdProperties, osd OSDI
 	podTemplateSpec.Spec.HostIPC = true
 
 	k8sutil.RemoveDuplicateEnvVars(&podTemplateSpec.Spec)
+	k8sutil.RemoveDuplicateTolerations(&podTemplateSpec.Spec)
```

Comment:
The original fix should be in ocs-operator, Rook should not remove duplicate toleartions , we discussed this in https://github.com/rook/rook/pull/15308

cc @BlaineEXE

Reply:
Thank you for the clarification. I understand that `NodeAffinity` comes from the StorageCluster CR, while the Toleration here is specific to the `CronJob` and works independently. This change only affects the `CronJob` pod spec and doesn’t impact tolerations at the `StorageCluster` level. Let me know if any adjustments are needed, and I can update the implementation accordingly.

Reply:
>CronJob
Why the cron job has duplicate toleartions? Ocs-operator should avoid adding it as per the latest discussion.

Reply:
I'll agree with Parth here. The entity creating `cephCluster` CR is some how appending duplicate tolerations in the `StorageClassDeviceSet`, which in turn gets propagated to the key rotation jobs. Rook is correctly appending all the tolerations provided by the user. 

So rather than removing duplicate entries in a separate method, a better approach, IMO, would be to investigate why duplicate tolerations were added to the StorageClassDeviceSet in the CephCluster CR in the first place.

Reply:
got it. I will close the PR